### DISCUSSION TITLE
y_preds.squeeze() In [18]: of "02. PyTorch Neural Network Classification"

### DISCUSSION THREAD

**Discussion by yabool2001** [mrdbourke/pytorch-deep-learning#471] [General]
Hi,

I've noticed that `y_preds.squeeze()` In [18]: of https://www.learnpytorch.io/02_pytorch_classification/ doesn't change anything. Idea was to get rid of extra dimension. But before `torch.Size` is ([5, 1]) and after `torch.Size` is ([5, 1]). Am I right or have I missed something?

**Comment by rwenen** [comment]
Hi @yabool2001
I use `tensor.squeeze()`, which can change the tensor's shape.
The following are the results before and after I used `tensor.squeeze()` to `y_preds`:

![image](https://github.com/mrdbourke/pytorch-deep-learning/assets/87791179/cc4ff437-ac19-481b-b1d6-b8044e67ae79)

![image](https://github.com/mrdbourke/pytorch-deep-learning/assets/87791179/ec270f81-3bc5-4207-a5ae-64cd31079402)

Hope to help you :)

**Reply by yabool2001** [reply]
Hi fooooooox,

Thx for interacting. I investigated your answer and made additional test. I see now that tensor.squeeze() can't change the tensor's shape. To be completely clear it can "returns a tensor with all specified dimensions of input of size 1 removed" (taken from pytorch docs). But it doesn't change tensor's shape. If you add

`print ( f"\n{y_preds=}" ) , print ( f"{y_preds.shape=}" )`

at the end of your code you will see that y_preds hasn't changed. The result will be:

`y_preds = tensor([[1.],
        [1.],
        [1.],
        [1.],
        [0.]], grad_fn=<RoundBackward0>)
y_preds.shape = torch.Size([5, 1])`

Here you can find my verification:
https://github.com/yabool2001/squeeze_test/blob/master/squeeze_test.ipynb

Look at `a_preds` tensor that is proper example of getting rid of extra dimension.

Looking forward to hearing your comments!

**Reply by rwenen** [reply]
I think I know what you mean
`y.squeeze()` will not change to `y` itself unless you overwrite itself
But I think can use `y.squeeze_()` which will in-place the original `y`
(https://pytorch.org/docs/stable/generated/torch.Tensor.squeeze_.html#torch.Tensor.squeeze_)
Following is an example of what I tried:

![image](https://github.com/mrdbourke/pytorch-deep-learning/assets/87791179/a68628dc-5f9b-4b61-8e92-01725aa47987)

![image](https://github.com/mrdbourke/pytorch-deep-learning/assets/87791179/208812c4-fac4-49f4-8c9b-f506c5c30da7)

Hope to help you :)

**Reply by yabool2001** [reply]
Great help,

`squeeze_()` removes the extra dimension. Like `y=y.squeeze()` does.

Thx again
